overlord/ifacestate/ifacemgr.go

```diff
@@ -42,7 +43,11 @@ type InterfaceManager struct {
 
 // Manager returns a new InterfaceManager.
 // Extra interfaces can be provided for testing.
-func Manager(s *state.State, extra []interfaces.Interface) (*InterfaceManager, error) {
+func Manager(s *state.State, hookManager *hookstate.HookManager, extra []interfaces.Interface) (*InterfaceManager, error) {
+	if hookManager != nil {
```

Comment:
If this is just for easier testing can you please add a comment saying that `// NOTE: hookManager is nil only when testing.`